lldb/docs/lldb-gdb-remote.txt

```diff
@@ -38,7 +38,44 @@ read packet: +
 read packet: $OK#9a
 send packet: +
 
+//----------------------------------------------------------------------
+// "QSupported"
+//
+// BRIEF
+//  Query the GDB remote server for features it supports
+//
+// PRIORITY TO IMPLEMENT
+//  Optional.
+//----------------------------------------------------------------------
 
+QSupported is a standard GDB Remote Serial Protocol packet, but
+there are several additions to the response that lldb can parse.
+An example exchange:
+
+send packet: qSupported:xmlRegisters=i386,arm,mips,arc;multiprocess+;fork-events+;vfork-events+
+
+read packet: qXfer:features:read+;PacketSize=20000;qEcho+;native-signals+;SupportedCompressions=lzfse,zlib-deflate,lz4,lzma;SupportedWatchpointTypes =aarch64-mask,aarch64-bas;
+
+In this example, three lldb extensions are reported:
+  PacketSize=20000
+    The base16 maximum packet size that the GDB Remote Serial stub
```

Comment:
"base 16"?

Also what is a serial stub? Does that literally mean when we use a serial port to talk to the debug server?

Reply:
I could say "the hex maximum packet size", I suppose, is that clearer?  One of the annoying things (IMO) about gdb remote serial protocol is that many numbers are hex numbers without a 0x prefix, but sometimes (probably mistakes) they're base 10.  So if you see a number like "2000" in a log file, you don't know what that number is.

The naming in this text file is a bit all over the place.  The protocol is called "gdb remote serial protocol".  The opposite side from lldb, debugserver or lldb-server, is called the stub.  Some places call it the "gdb RSP stub", some just say "stub".

Reply:
(incidentally debugserver also reports a compression minimum size -- which I'll be removing after this patch is landed because it does nothing -- and that number is sent as a base 10 number.  lol it's a wild west for numbers.)